wordstats/cognate_info.py

```diff
@@ -0,0 +1,186 @@
+import os
+from datetime import datetime
+
+from sqlalchemy import Table
+import configparser
+from .word_info import WordInfo, UnknownWordInfo
+from .utils.mem_footprint import total_size
+from .base_service import BaseService, Base
+from .config import MAX_WORDS
+from .metrics_computers import *
+from .cognate_files_path import *
+from .getchunix import read_single_keypress
+
+
+class CognateInfo(object):
+    def __init__(self, language_ids, method_name):
+        self.language_ids = language_ids
+        self.whitelist = []
+        self.candidates = []
+        self.blacklist = []
+        self.rules = []
+        self.method_name = method_name
+
+    def all_blacklist(self):
+        return self.blacklist.items()
+
+    def all_whitelist(self):
+        return self.whitelist.items()
+
+    def all_rules(self):
+        return self.rules.items()
+
+    def __getitem__(self, key):
+        return self.get(key)
+
+
+    # also save new dicts
+    @classmethod
+    def load_from_path(cls, language_ids, method_name= ""):
+        """
+        Loads cognate information given the cognate code (e.g. ennl) and method_name
+        Loads the word info from the given file
+        where the format expected are two words per line separated by a whitespace e.g.
+
+        die die
+        was was
+        er er
+        in in
+        ja ja
+        ...
+
+        :param cognate_code:
+        :param method_name:
+        :return:
+        """
+
+        new_registry = cls(language_ids, method_name)
+
+        language_code_path = path_of_cognate_whitelist(language_ids)
+        new_registry.whitelist = load_from_path(language_code_path).splitlines()
+
+        language_code_path = path_of_cognate_blacklist(language_ids)
+        new_registry.blacklist = load_from_path(language_code_path).splitlines()
+
+        language_code_path = path_of_cognate_candidates(language_ids, method_name)
+        new_registry.candidates = load_from_path(language_code_path).splitlines()
+
+        return new_registry
+
+    def save_candidates(self):
+
+        language_code_path = path_of_cognate_candidates(self.language_ids, self.method_name)
+        save_to_file(language_code_path, '\n'.join(self.candidates))
+
+    def save_evaluation(self):
+
+        language_code_path = path_of_cognate_whitelist(self.language_ids)
+        save_to_file(language_code_path, '\n'.join(self.whitelist))
+
+        language_code_path = path_of_cognate_blacklist(self.language_ids)
+        save_to_file(language_code_path, '\n'.join(self.blacklist))
+
+    # generates candidates based on distance function func and word lists
+    def apply_distance_metric(self, wordlist1, wordlist2, func):
+
+        for w1 in wordlist1:
+            newlist = list(filter(lambda x: len(x)>0 , map(lambda w: w1 + ' ' + w if func(w1, w) else "", wordlist2)))
+            self.candidates.extend(newlist)
+
+    # note: only executable in terminal, reactive to one-key stroke
+    def start_quiz(self):
```

Comment:
one thing that would be good is to move all the stuff related to the quiz into a separate module. 
if you think about it, the cognates class can work independent of the way the blacklist / whitelists are generated.
all it matters is that they exist from the pov of the CognateInfo class. agreed?